b2share/modules/schemas/cli.py

```diff
@@ -46,3 +64,225 @@ def init(verbose):
         load_root_schemas(cli=True, verbose=verbose)
     except RootSchemaAlreadyExistsError as e:
         raise click.ClickException(str(e))
+
+            
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.argument('community')
+@click.argument('name')
+def block_schema_add(verbose, community, name):
+    """Adds a block schema to the database. Community is the ID or NAME of the maintaining community for this block schema. Name is the name as displayed in block_schema_list command."""
+    if verbose:
+        click.secho("Creating block schema with name %s to be maintained by community %s" % (name, community))
+    comm = get_community_by_name_or_id(community)
+    if not comm:
+        raise click.BadParameter("There is no community by this name or ID: %s" % community)
+    if len(name)>255:
+        raise click.BadParameter("NAME parameter is longer than the 255 character maximum")
+    block_schema = BlockSchema.create_block_schema(comm.id, name)
+    db.session.commit()
+    if verbose:
+        click.secho("Created block schema with name %s and id %s" % (name, block_schema.id))
+
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.option('-c','--community', help='show only block schemas filtered by maintaining community id or name')
+def block_schema_list(verbose, community):
+    """Lists all block schemas for this b2share instance (filtered for a community)."""
+    comm = None
+    if community:
+        comm = get_community_by_name_or_id(community)
+    community_id = None
+    if comm:
+        community_id = comm.id
+    if verbose:
+        click.secho("filtering for community %s" % comm)
+    try:
+        block_schemas = BlockSchema.get_all_block_schemas(community_id=community_id)
+    except BlockSchemaDoesNotExistError:
+         raise click.ClickException("No block schemas found, community parameter was: %s" % community)
+    click.secho("BLOCK SCHEMA ID\t\t\t\tNAME\t\tMAINTAINER\tDEPRECATED\t#VERSIONS")
+    for bs in block_schemas:
+        bs_comm = Community.get(id=bs.community)
+        click.secho("%s\t%s\t%s\t%s\t\t%d" % (bs.id,bs.name[0:15].ljust(15), bs_comm.name[0:15].ljust(15), bs.deprecated, len(bs.versions)  ))
+        
+@schemas.command()
+@with_appcontext
+@click.option('-v','--verbose', is_flag=True, default=False)
+@click.option('-n','--name',help='set the name of the community')
+@click.option('-c','--community',help='set the maintaining community by name or id')
+@click.option('-d','--deprecated',help='(un)set deprecated bit, 1 is deprecated, 0 is not deprecated')
```

Comment:
Rather than `bit` which is an implementation detail. We could just use the term `flag` which people tend to understand better.

The documentation would then be: "(un)mark the schema as deprecated. Values can be either `True` to add the deprecation flag, or `False` to remove the flag."